youtube-client-app/src/app/app.component.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  Component, ElementRef, OnInit, ViewChild,
+} from '@angular/core';
+import { IResponse } from './common/models/search-response.models';
+import youtubeResponse from './common/mocks/response';
+import { ISearchItem } from './common/models/search-items.models';
+import { VideoSearchService } from './services/video-search.service';
+import { ISortSettings, sortMap } from './common/models/sort-settings.model';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.scss'],
+})
+export class AppComponent implements OnInit {
+  @ViewChild('searchResult') searchListRef?: ElementRef;
+
+  public title: string = 'youtube-client-app';
+
+  public youtubeResponse: IResponse = youtubeResponse;
+
+  public items: ISearchItem[] = youtubeResponse.items;
+
+  public settingActive: boolean = false;
+
+  public keyword: string = '';
+
+  public sortSettings: ISortSettings = {
+    filterBy: sortMap.empty,
+    sortIncrease: true,
+    keywords: '',
+  };
+
+  constructor(private readonly videoSearchService: VideoSearchService) {}
+
+  ngOnInit(): void {
+    this.videoSearchService.searchVal$.subscribe((searchVal) => this.onSearch(searchVal));
+  }
+
+  onSearch(searchVal: string) {
+    (this.searchListRef as ElementRef).nativeElement.style.display = 'block';
```

Comment:
why did you implement it that way instead of using ngIf?

Reply:
Refractored this whith ngIf